test/index.js

```diff
@@ -124,4 +141,87 @@ describe('Drip', function(){
         .end(done);
     });
   });
+
+  describe('.setLimit()', function() {
+    it('should set limit succefully', function(done) {
+      var appId = '140286';
+      var key = ['drip', appId].join(':');
+      drip.settings.account = appId;
+
+      var remaining = Math.floor((Math.random() * 1000) + 1);
+      var headers = {'x-ratelimit-remaining': remaining};
+
+      drip.setLimit(headers, function() {
+        db.get(key, function(err, res) {
+          assert.deepEqual(err, null);
+
+          limit = JSON.parse(res);
+          assert.deepEqual(limit.remaining, remaining);
+          done();
+        });
+      });
+    });
+  });
+
+  describe('.limit()', function() {
+    it('req is called when there is no limit in redis', function(done) {
+      var appId = '300888';
+      drip.settings.account = appId;
+
+      req = function() {
+        done();
+      };
+
+      drip.limit(req, null);
+    });
+
+    it('req is called when there is remaining', function(done) {
+      var appId = '201187';
+      drip.settings.account = appId;
+
+      var remaining = 42;
+      var headers = {'x-ratelimit-remaining': remaining};
+
+      drip.setLimit(headers, function() {
+        req = function() {
+          done();
+        };
+        drip.limit(req, null);
+      });
+    });
+
+    it('req is called when reset is passed', function(done) {
+      var appId = '201187';
+      drip.settings.account = appId;
+
+      var remaining = 0;
+      var headers = {
+        'x-ratelimit-remaining': remaining,
+        'x-ratelimit-reset': Date.now()
```

Comment:
we should subtract a second here to avoid sporadic failures when it runs too fast